synapse/federation/sender/per_destination_queue.py

```diff
@@ -180,6 +192,16 @@ def attempt_new_transaction(self) -> None:
             logger.debug("TX [%s] Transaction already in progress", self._destination)
             return
 
+        if not self._federation_shard_config.should_send_to(
+            self._instance_name, self._destination
+        ):
```

Comment:
How hot of a path is this? The status of this shouldn't change after construction, correct? Should we make this a flag on the object instead.